src/immut_helloworld.ts

```diff
@@ -74,3 +86,100 @@ export class HelloReply {
 		return this.underlying.serializeBinary();
 	}
 }
+
+export enum PersonContactEnum {
+	NOT_SET = 0,
+	EMAIL = 2,
+	MOBILE = 3,
+}
+
+export class Person {
+	private underlying: helloworld.Person;
+
+	constructor() {
+		this.underlying = new helloworld.Person();
+	}
+
+	static fromPerson(u: helloworld.Person): Person {
+		let ihr = new Person();
+		ihr.underlying = u.cloneMessage();
+		return ihr;
+	}
+
+	static Deserialize(buffer: Uint8Array): Person {
+		return Person.fromPerson(helloworld.Person.deserializeBinary(buffer));
+	}
+
+	SetName(name: string): Person {
+		if (this.underlying.getName() === name) {
+			return this;
+		} else {
+			let instance = Person.fromPerson(this.underlying.cloneMessage());
+			instance.underlying.setName(name);
+			return instance;
+		}
+	}
+
+	get Name(): string {
+		return this.underlying.getName();
+	}
+
+	SetMobile(num: Uint64): Person {
+		/* This logic is to ensure that if underlying value has not changed
+			* return the same instance */
+		if (this.underlying.getMobile() === num.toNumber()) {
+			return this;
+		} else {
+			let instance = Person.fromPerson(this.underlying.cloneMessage());
+			instance.underlying.setMobile(num.toNumber());
+			return instance;
+		}
+	}
+
+	/* This logic handles the clearing of field in the oneof case */
```

Comment:
And provides clear semantic separation between normal fields and fields from a `oneof` group